Source/Core/Core/HW/EXI/EXI.cpp

```diff
@@ -39,32 +39,64 @@ static void UpdateInterruptsCallback(u64 userdata, s64 cycles_late);
 
 namespace
 {
-void AddMemoryCards(int i)
+void AddMemoryCards(Slot slot)
 {
-  TEXIDevices memorycard_device;
+  EXIDeviceType memorycard_device;
   if (Movie::IsPlayingInput() && Movie::IsConfigSaved())
   {
-    if (Movie::IsUsingMemcard(i))
+    if (Movie::IsUsingMemcard(slot))
     {
-      if (Config::Get(Config::GetInfoForEXIDevice(i)) == EXIDEVICE_MEMORYCARDFOLDER)
-        memorycard_device = EXIDEVICE_MEMORYCARDFOLDER;
+      if (Config::Get(Config::GetInfoForEXIDevice(slot)) == EXIDeviceType::MemoryCardFolder)
+        memorycard_device = EXIDeviceType::MemoryCardFolder;
       else
-        memorycard_device = EXIDEVICE_MEMORYCARD;
+        memorycard_device = EXIDeviceType::MemoryCard;
```

Comment:
I'm not sure.  I think the purpose here is to force it to be a memory card (and theoretically directly using the configured value could result in something other than a memory card); the Movie/TAS code only supports indicating whether a memory card is in use, and doesn't support other devices.  But if we're overriding settings for movie purposes, why bother supporting both raw memory cards and memory card folders?  Someone more experienced with the movie functions would have to chime in here.

Reply:
Looking at it further, I think the best approach would be to get rid of `AddMemoryCards` entirely and then handle this in `MovieConfigLoader` which is injected into the config system (CC @AdmiralCurtiss).  That wasn't possible until recently when the EXI and other settings were moved to the new config system.

JosJuice says that there is a benefit to supporting both raw memory cards and memory card folders, since users need to manually supply the save file (it's not included in the move file).  On the other hand, devices other than memory cards aren't supported and won't sync at all (at least for things like the microphone).  So forcing it to be a memory card works, but I've changed it to instead stick with the user-specified device and generate a panic alert if an invalid device is selected.

Reply:
Yes, I would agree, handling this via `LoadFromDTM()` in `MovieConfigLoader.cpp` is likely the best approach here.